pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -261,7 +261,8 @@ pub mod pallet {
 	#[pallet::call]
 	impl<T: Config> Pallet<T>
 	where
-		<CurveParameterTypeOf<T> as Fixed>::Bits: Copy + ToFixed + AddAssign + BitOrAssign + ShlAssign + TryFrom<u128>,
+		<CurveParameterTypeOf<T> as Fixed>::Bits: Copy + ToFixed + AddAssign + BitOrAssign + ShlAssign + TryFrom<U256>,
+		CollateralCurrenciesBalanceOf<T>: Into<U256> + TryFrom<U256>,
```

Comment:
Using bigger integers is also a better fix for the overflow issues we had when computing the refund amounts. This second trait bound is for this fix.

Reply:
We can also make this type configurable, it just needs to have twice the size of the balance type.

Reply:
No strong opinion on this. I don’t think any other project is using a balance type larger than u128.

Reply:
Good thing is you can also configure it to be smaller, avoiding blowing it up to a U256 when it's completely unnecessary, as it would be if your balance is just a u64 (where a u128 would be enough)

Reply:
Added a TODO in cffd4bd2ae5a18d02722e18abdd6c0d14dc497f8